yazi-cli/src/main.rs

```diff
@@ -46,6 +48,15 @@ async fn main() -> anyhow::Result<()> {
 				package::Package::add_to_config(repo).await?;
 			}
 		}
+
+		Command::Sub(cmd) => {
+			yazi_dds::init();
+			let kinds = cmd.kinds.split(',').map(|s| s.to_owned()).collect::<HashSet<_>>();
```

Comment:
```suggestion
			let kinds: HashSet<_> = cmd.kinds.split(',').collect();
```

This avoids reallocating memory.

Reply:
I agree, although this also changes the type to `HashSet<&str>`, and in this case a `HashSet<String>` is required. This is due to the type of `abilities` here 
https://github.com/sxyazi/yazi/blob/9cd3ee0cd702941ec62edf7fd8d637e9dc0a5ac0/yazi-dds/src/body/hi.rs?plain=1#L18

I think this means the data will be copied anyway. It might be possible to use `.split_off` which consumes the input string, but do you think it's worth it? In this case the input will always be very small.

Reply:
Should be fixed in https://github.com/sxyazi/yazi/pull/1004/commits/8cb34fd0a871955fbb09db0fcbe228e908761383